src/share/classes/com/sun/javatest/regtest/exec/Agent.java

```diff
@@ -927,6 +986,8 @@ private static String getKey(File dir, JDK jdk, List<String> vmOpts) {
         private float timeoutFactor = 1.0f;
         private int maxPoolSize;
         private Duration idleTimeout;
+        // default is 2 i.e. we retry a failed agent selection once
+        private int numAgentSelectionAttempts = 2;
```

Comment:
The default should be 1, meaning that the new behavior is opt-in, not opt-out.

You can always set a different default in the layers of software wrapping `jtreg`.

Reply:
I've updated the PR to use the default as 1.